charts/rancher-pushprox/rancher-pushprox/0.1.300/templates/pushprox-clients.yaml

```diff
@@ -1,18 +1,33 @@
+{{- if .Values.clients.deployment.replicas }}
+  {{- if empty .Values.clients.deployment.affinity }}
```

Comment:
nit: We can make combine this statements

Reply:
The condition will be super complex if we want to combine those two lines because 2 of 4 cases are what we want.

<table>
<tr>
	<td> replicas
	<td> affinity
	<td> action
<tr>
	<td> is 0
	<td> empty
	<td> pass
<tr>
	<td> is 0
	<td> not empty
	<td> error out
<tr>
	<td> not 0
	<td> empty
	<td> error out
<tr>
	<td> not 0
	<td> not empty
	<td> pass
</table>

For example, a way to express it is 

{{- if not or (and .Values.clients.deployment.replicas .Values.clients.deployment.affinity) (and  (not .Values.clients.deployment.replicas) (not .Values.clients.deployment.affinity) ) }}
{{- fail ".Values.clients.deployment.affinity can not be empty. Please set either nodeAffinity, inter-pod affinity or anti-affinity." }}
{{- end }}

@nickgerace  And idea how we can simplify the above expression?

Reply:
Do we need to validate that the user provides affinity rules correctly for Deployments? I think it should be fine to say "use at your own risk, here be dragons" if we call it out as an advanced feature (see comment below): https://github.com/rancher/charts/pull/1227#discussion_r645805780

Reply:
@aiyengar2  You are right. 
I will leave it to the user to make sure the affinity rules is not empty. Also, there are default values in the value.yaml .